### PR TITLE
Register Jackson parameter names module

### LINKED ISSUE #1852: Should jackson ParameterNamesModule be supplied?
As part of the uplift to Java 8 for 1.x should the jackson parameter names module have been supplied by default?

https://github.com/FasterXML/jackson-modules-java8/tree/master/parameter-names

### PR DESCRIPTION
**Pull request by nickbabcock**
Closes #1852

Addresses @arteam's concern about users who don't compile with `-parameters`, as only the `dropwizard-jackson` test module is compiled with `-parameters` and the other modules don't fail.

### CODE REVIEW

File: dropwizard-jackson/src/test/java/io/dropwizard/jackson/ParanamerModuleTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
```
Comment (joschi) [inline dropwizard-jackson/src/test/java/io/dropwizard/jackson/ParanamerModuleTest.java]: Wrong assertion?

**Comment by nickbabcock** [comment]
oops 😊 

Good catch, fixed.

File: dropwizard-jackson/pom.xml [hunk 2]
```diff
@@ -73,4 +77,25 @@
...
+                <artifactId>maven-compiler-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>default-testCompile</id>
+                        <phase>test-compile</phase>
+                        <goals>
+                            <goal>testCompile</goal>
+                        </goals>
+                        <configuration>
+                            <compilerArgument>-parameters</compilerArgument>
```
Comment (arteam) [inline dropwizard-jackson/pom.xml]: I believe Maven does that by default. But I'm not sure, will try look it up.

Comment (nickbabcock) [inline dropwizard-jackson/pom.xml]: I commented out that line and the tests fails (locally), so it doesn't look like a default.

Comment (joschi) [inline dropwizard-jackson/pom.xml]: Same for me here. I think this is the correct way to enable retrieving method parameters via reflection.

Reference: https://docs.oracle.com/javase/tutorial/reflect/member/methodparameterreflection.html

**Comment by nickbabcock** [comment]
Updated release notes in 95c0468

**Pull request merged**